internal/provider/incident_catalog_type_attribute_resource_test.go

```diff
@@ -136,3 +137,45 @@ func testAccIncidentCatalogTypeAttributeResourceConfig(attribute client.CatalogT
 
 	return buf.String()
 }
+
+func TestAttributeToPayload_PreservesModes(t *testing.T) {
```

Comment:
this is unfortunately the best test I could come up with. I tried to write a proper acceptance test, but couldn't find a good way of updating an attribute to make it internal through the public API that's consistent with how it works via the internal API when setting the team type in settings